### ISSUE TITLE
NER model not learning

### ISSUE THREAD

**Issue by JiajunBao** [zalandoresearch/flair#1144]
I am training a NER model on chemistry documents. The model does not learn any patterns.
I format my data in this way:
![image](https://user-images.githubusercontent.com/24786001/65394819-55f18000-dd60-11e9-9678-1990c2dd2710.png)
The first column is the word token and the second column is the IOB tags. (Most of those tags are O)
I set the tagger and embedding in this way:
![image](https://user-images.githubusercontent.com/24786001/65394896-18412700-dd61-11e9-98b9-1be45e43470d.png)
I am using GloVe embedding. Words in my documents are mostly included in the dictionary of GloVe.
The model simply assigns 'O' tags to every word in the test set.
I have tried:
- adding CharacterEmbeddings
- set a smaller learning rate (say 0.1, 0.001)

If anyone could give me some suggestions, that would be greatly appreciated.

**Comment by alanakbik** [comment]
@JiajunBao can you paste the full training script?

**Comment by JiajunBao** [comment]
```
from flair.data import Corpus
from flair.datasets import ColumnCorpus
from flair.embeddings import TokenEmbeddings, WordEmbeddings, StackedEmbeddings
from flair.models import SequenceTagger
from flair.training_utils import EvaluationMetric
from flair.trainers import ModelTrainer
from flair.visual.training_curves import Plotter
from typing import List
from pathlib import Path
import os

ROOT_FOLDER = Path().resolve()

def main():
    # 1. get the corpus
    # define columns
    columns = {0: 'text', 1: 'ner'}

    # this is the folder in which train, test and dev files reside
    DATA_FOLDER = os.path.join(ROOT_FOLDER, 'data', 'processed', 'sentences')

    # init a corpus using column format, data folder and the names of the train, dev and test files
    corpus: Corpus = ColumnCorpus(DATA_FOLDER, columns,
                                  train_file='train.sentence.csv',
                                  test_file='test.sentence.csv',
                                  dev_file='valid.sentence.csv')

    print('size of corpus is: {}/{}/{}'.format(len(corpus.train), len(corpus.dev), len(corpus.test)))

    # 2. what tag do we want to predict?
    tag_type = 'ner'

    # 3. make the tag dictionary from the corpus
    tag_dictionary = corpus.make_tag_dictionary(tag_type=tag_type)
    print(tag_dictionary.idx2item)

    # 4. initialize embeddings
    embedding_types: List[TokenEmbeddings] = [

        WordEmbeddings('glove'),

        # comment in this line to use character embeddings
        # CharacterEmbeddings(),

        # comment in these lines to use flair embeddings
        # FlairEmbeddings('news-forward'),
        # FlairEmbeddings('news-backward'),
    ]

    embeddings: StackedEmbeddings = StackedEmbeddings(embeddings=embedding_types)

    # 5. initialize sequence tagger


    tagger: SequenceTagger = SequenceTagger(hidden_size=256,
                                            embeddings=embeddings,
                                            tag_dictionary=tag_dictionary,
                                            tag_type=tag_type,
                                            use_crf=True,
                                            )

    # 6. initialize trainer
    trainer: ModelTrainer = ModelTrainer(tagger, corpus)

    # 7. start training
    OUTPUT_FOLDER = os.path.join(ROOT_FOLDER, 'models', 'sentence_checkpoints')
    trainer.train(OUTPUT_FOLDER,
                  learning_rate=0.1,
                  mini_batch_size=32,
                  max_epochs=80,
                  anneal_factor=0.9,
                  monitor_train=True,
                  monitor_test=True,
                  weight_decay=0.99,
                  embeddings_storage_mode='none')

    # # 8. plot weight traces (optional)
    plotter = Plotter()
    plotter.plot_training_curves(os.path.join(OUTPUT_FOLDER, 'loss.tsv'))
    plotter.plot_weights(os.path.join(OUTPUT_FOLDER, 'weights.txt'))

if __name__ == '__main__':
    main()
```

**Comment by alanakbik** [comment]
@JiajunBao could you try removing the `weight_decay` option?

**Comment by JiajunBao** [comment]
It works now! Thank you.
